src/main/java/com/intel/gkl/smithwaterman/IntelSmithWaterman.java

```diff
@@ -103,7 +110,7 @@ public SWNativeAlignerResult align(byte[] refArray, byte[] altArray, SWParameter
         return new SWNativeAlignerResult(new String(cigar).trim(), offset);
     }
 
-    public int getStrategy(SWOverhangStrategy strategy)
+    public int getStrategy(SWOverhangStrategy strategy) //TODO: Check with GKL Team
```

Comment:
We can add null check for strategy here

Reply:
I thought about default strategy, do we have strategy with id 0? In GetStrategy function is initial value for  intStrategy.

Reply:
Reviewing the code it seems getStraegy should be a private function. I did not find it in GATK anywhere and it is only use in in this class.   GKL seems to remap the Overhang Strategy type for some reason. 

In SWNativeAlignerResult we need to check either overhangStrategy or the modified  intStrategy to be a valid input before we call  alignNative. 

Do not call  alignNative if it is not a valid strategy and return an error/exception.

Reply:
@mkuligow  Please add default to throw  IllegalArgumentException here.